### ISSUE TITLE
(design) Replace Patient Checkins with Patient Visits

### ISSUE THREAD

**Issue by jniles** [Vanga-Hospital/bhima-2.X#23]
This issue concerns the design for Patient Visits.  It supersedes both https://github.com/IMA-WorldHealth/bhima-2.X/issues/652 and https://github.com/IMA-WorldHealth/bhima-2.X/issues/653.

Patient Checkins do not provide enough information to really be useful in practice.

### Patient Visits
This feature allows a patient to be *admitted* and then *discharged* at the end of their visit.  Both the patient admission and discharge can have notes.  An [ICD-9](https://en.wikipedia.org/wiki/List_of_ICD-9_codes) diagnosis is required on discharge.

#### UI/UX
![no patient visits](https://cloud.githubusercontent.com/assets/896472/21422700/2ae6aa5a-c839-11e6-9135-875f58d16c28.png)
_Fig 1: Patient Visits - No Visits Yet_

![patient visits completed](https://cloud.githubusercontent.com/assets/896472/21422698/2ad5b86c-c839-11e6-915b-049fe0a56ef6.png)
_Fig 2: Patient Visits - Three Visits Completed_

![patient visits in progress](https://cloud.githubusercontent.com/assets/896472/21422695/2a8950ee-c839-11e6-8ea8-3d6fbb2320cd.png)
_Fig 3: Patient Visits - Visit In Progress_

![patient visits admission](https://cloud.githubusercontent.com/assets/896472/21422699/2ada6cd6-c839-11e6-8e6b-82c13922e7b9.png)
_Fig 4: Patient Visits - Admission Summary_

![patient visits discharge](https://cloud.githubusercontent.com/assets/896472/21422697/2a95bb4a-c839-11e6-85ef-32dfc33f05a9.png)
_Fig 5: Patient Visits - Discharge Summary_

**Comment by jniles** [comment]
I'll grab this one.

**Comment by jniles** [comment]
The following changes have been requested:
 1. Have two separate diagnosis fields for admission diagnosis and discharge diagnosis.  These should come from the [ICD10 medical codes](http://www.icd10data.com/).  The above incorrectly states these will be IDC-9 codes, which are outdated.

The proposed database schema looks like this:
```sql
CREATE TABLE `patient_visit` (
  `uuid` BINARY(16) NOT NULL,
  `patient_uuid` BINARY(16) NOT NULL,
  `start_date` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `end_date` datetime DEFAULT NULL,
  `start_notes` TEXT,
  `end_notes` TEXT,
  `start_diagnosis_id` INT(10) unsigned,
  `end_diagnosis_id` INT(10) unsigned,
  `user_id` smallint(5) unsigned NOT NULL,
  PRIMARY KEY (`uuid`),
  UNIQUE KEY `patient_visit_1`(`patient_uuid`, `start_date`, `end_date`),
  KEY `patient_uuid` (`patient_uuid`),
  KEY `user_id` (`user_id`),
  KEY `start_diagnosis_id` (`start_diagnosis_id`),
  KEY `end_diagnosis_id` (`end_diagnosis_id`),
  FOREIGN KEY (`patient_uuid`) REFERENCES `patient` (`uuid`) ON DELETE CASCADE ON UPDATE CASCADE,
  FOREIGN KEY (`user_id`) REFERENCES `user` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
  FOREIGN KEY (`start_diagnosis_id`) REFERENCES `icd10` (`id`) ON UPDATE CASCADE,
  FOREIGN KEY (`end_diagnosis_id`) REFERENCES `icd10` (`id`) ON UPDATE CASCADE
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```

**Comment by jniles** [comment]
Closed in #93.
